namer/namer.cc

```diff
@@ -726,58 +701,6 @@ class SymbolDefiner {
         emitRedefinedConstantError(ctx, errorLoc, symbol.name(ctx), symbol.kind(), prevSymbol);
     }
 
-    core::ClassOrModuleRef ensureScopeIsClass(core::MutableContext ctx, core::SymbolRef scope,
-                                              core::NameRef nameForErrors, core::LocOffsets loc) {
-        // Common case: Everything is fine, user is trying to define a symbol on a class or module.
-        if (scope.isClassOrModule()) {
-            ENFORCE(!isMangleRenameUniqueName(ctx, scope.name(ctx)));
-            return scope.asClassOrModuleRef();
-        }
-
-        // Check if class was already mangled.
-        auto klassSymbol = ctx.state.lookupClassSymbol(scope.owner(ctx).asClassOrModuleRef(), scope.name(ctx));
-        if (klassSymbol.exists()) {
-            ENFORCE(isMangleRenameUniqueName(ctx, klassSymbol.data(ctx)->name));
-            if (auto e = ctx.beginError(loc, core::errors::Namer::InvalidClassOwner)) {
```

Comment:
I guess a followup PR can comment out this error at its definition?

Reply:
What do you mean?

This error no longer happens, because the "can't nest under" error never happens. The scope of a class def or const assign is now always a class symbol by construction.

What happens is that _other_ code starts to error, saying that you can't initialize a class or module by constant assignment:

```ruby
X = 1 # <- new error is here ("can't initialize class `X` by assignment")
X::Y = 1 # <- old error used to be here ("can't nest Y under static field X")
```

Reply:
There are some scripts that suppress `4015`. I'd rather not have to figure out a way to update that in a backwards-compatible way, so I think I'll just leave the error code around but unused, if you don't mind.